llvm/lib/Transforms/InstCombine/InstCombineVectorOps.cpp

```diff
@@ -476,7 +476,8 @@ Instruction *InstCombinerImpl::visitExtractElementInst(ExtractElementInst &EI) {
   // it may make the operand poison.
   BinaryOperator *BO;
   if (match(SrcVec, m_BinOp(BO)) && cheapToScalarize(SrcVec, Index) &&
-      (HasKnownValidIndex || isSafeToSpeculativelyExecute(BO))) {
+      (HasKnownValidIndex ||
+       isSafeToSpeculativelyExecuteWithVariableReplaced(BO))) {
```

Comment:
Why is it necessary here? Isn't this transform change BO to handle strictly a subset of what it was previously?

Reply:
See the comment above this code: We may replace the operand with poison if the extractelement index is out of bounds.